c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -578,6 +662,84 @@ public:
   }
 };
 
+class HttpOverCapnpFactory::HttpOverCapnpConnectResponseImpl : public kj::HttpService::ConnectResponse {
+public:
+  HttpOverCapnpConnectResponseImpl(
+      HttpOverCapnpFactory& factory, capnp::HttpService::ConnectParams::Reader request,
+      capnp::HttpService::ConnectClientRequestContext::Client context,
+      kj::Own<kj::AsyncInputStream> upStream, kj::Own<kj::AsyncIoStream> httpProxyStream) :
+      context(context), request(request), factory(factory), upStream(kj::mv(upStream)),
+      httpProxyStream(kj::mv(httpProxyStream)) {}
+
+  void accept(uint statusCode, kj::StringPtr statusText, const kj::HttpHeaders& headers) override {
+    KJ_REQUIRE(statusCode >= 200 && statusCode < 300, "the statusCode must be 2xx for accept");
```

Comment:
In theory it's not really necessary to validate the status code in http-over-capnp. It's necessary in the actual HTTP protocol implementation, where status codes matter. But in http-over-capnp, it's already possible to distinguish success and error cases based on how the protocol is used, independently of the status code. http-over-capnp is really an RPC version of the `kj::HttpService` interface, _not_ of the actual HTTP protocol, and as a result it can actually express things that wouldn't be possible in real HTTP, and that's OK. We don't actually validate status codes anywhere else in http-over-capnp, we just pass them through.

So, I would remove these checks.